### ISSUE TITLE
Question: Why do webiny builds the libraries with babel, instead of tsc?

### ISSUE THREAD

**Issue by latotty** [webiny/webiny-js#1248]
## This is:

- Question

## Detailed Description
Based on this issue: https://github.com/webiny/webiny-js/issues/1247

Currently webiny builds all the libraries with babel-typescript, instead of simply typescript.
In my view this is one unnecessary step, which slows the build a bit. Also creates the problem stated in #1247 .
Typescript can compile react and node libraries also, and when the final deploy build is created (with webpack / babel-typescript) the result will be the same.

So I would like to understand the decision behind it.

**Comment by doitadrian** [comment]
This is really a question for @Pavel910. But, his availability will be limited a bit today, so he won't respond soon, but I'm sure he'll do that as soon as he gets the chance.

The idea you're proposing here sounds interesting, to me at least, because I'm sure TSC out of the box probably already supports everything one might need. And it's also great it can be combined with webpack.

But again, let's wait for @Pavel910's response. I'm also interested in hearing his thoughts.

**Comment by Pavel910** [comment]
@latotty @doitadrian the main reason is build speed while developing. `tsc` is very slow because it does all the type checks. Let me elaborate:

When working on, say, `app-page-builder` in our repository, you run `admin` app which imports `@webiny/app-page-builder`. `admin` app itself doesn't build packages, it only consumes them via `node_modules`. So those packages need to be ready for use. If you open `package.json` file of any `app-*` package in our repo, you'll find 2 scripts: `build` and `watch`. `build` is used when building packages, and you'll find a `postbuild` which runs `tsc`; that's what we need for production-ready builds that will end up on `npm`. When releasing packages, we don't care if build takes 1 or 15 seconds.

But when developing a React app or API packages locally - it's very important. You want fast rebuilds and reloads to keep development iterations fast. That's why we have that `watch` script - it only transpiles the code using `babel` and strips away TS. That caters for fast rebuilds. 

**But what about type checking?** - that's why we have 2 **tsconfig** files: 

- **tsconfig.json** is used by the IDE (we all use Webstorm so it `just works`), which checks your types independently of the `watch` process and you still have your TS reporting
- **tsconfig.build.json** which is only used during the `build` script, which performs full build + type checking

If we only relied on `tsc`, every file change would take 10 seconds to be compiled + type-checked - and your frustration would go up by the minute.

This is the best we managed to do, and I've seen other projects do the same, for exact same reasons. 
If there is a better way - I'm all 👂. If we can simplify things - I'm all for it.
